src/main/java/org/jabref/gui/preferences/PreferencesFilterDialog.java

```diff
@@ -22,6 +26,7 @@
 
     private final JabRefPreferencesFilter preferencesFilter;
     private final ObservableList<JabRefPreferencesFilter.PreferenceOption> preferenceOptions;
+    private final List<JabRefPreferencesFilter.PreferenceOption> auxillaryPreferenceOptions;
```

Comment:
If you use the `FilteredList`, then the code gets somewhat cleaner (as you don't need to manage two lists on your own). Have a look here:
https://github.com/JabRef/jabref/blob/b39ed7b94108308fb078adcd07e28e52d4a3df48/src/main/java/org/jabref/gui/maintable/MainTableDataModel.java#L29-L33

Reply:
Thank you for the suggestions! I have made the necessary changes.